mathics/main.py

```diff
@@ -442,6 +443,8 @@ def dump_tracing_stats():
             definitions.set_line_no(0)
         elif not args.execute:
             return exit_rc
+    else:
+        definitions.set_inputfile("")
```

Comment:
An alternative would be to leave the attribute undefined here. Then, in `definitions.get_inputfile` just check if the attribute is already set with  `hasattr(self, "inputfile")`. and if not, just return "".

Reply:
Yes, it seems better. I have just added it and removed the env var.